### PR TITLE
initial v2 api migration

### PR DESCRIPTION
**Pull request by rajan-si**

### CODE REVIEW

File: Amp.js [hunk 1]
```diff
@@ -1,69 +1,100 @@
...
+ * @param  {int} timeOutMilliseconds - request timeOut in milliSeconds default 10 seconds
+ * @param  {int} sessionLifeTimeSeconds - session life time in Seconds default 30 minutes
+ * @param  {boolean} dontUseTokens - flag to indicate either generate amptoken automatically or use the custom token.
  *
  */
-module.exports = class Amp {
-  constructor(options = {}) {
-    this.key = options.key;
-    if (!this.key) throw new Error('Project Key Needed!');
+module.exports  = class AmpV2 {
```
Comment (steve-si) [inline Amp.js]: Why are you changing the shared API?

File: Amp.js [hunk 2]
```diff
@@ -1,69 +1,100 @@
...
  * @constructor
- * @param {Object} options
- *
- * Options:
- *  key - project key
- *  domain - domain of Amp server
- *  apiPath - path of API
- *  sessionTTL - TTL for session
- *  timeout - TTL for requests
+ * @param  {string} key - project key
```
Comment (steve-si) [inline Amp.js]: Why are you changing the interface?  The interface needs to adhered to the common spec.

File: Amp.js (hunk 1 above)
Comment (rajan-si) [reply Amp.js]: This was a mistake. I forgot to make the change to Amp. Fixed right now.

**Review by steve-si — CHANGES_REQUESTED** [review]
I think you forgot to use the observeUrl property

File: Amp.js [hunk 3]
```diff
@@ -1,69 +1,68 @@
...
+    }
+    this.key = key;
 
-    // the Session Constructor
-    const _this = this;
-    this.Session = function(sessionOptions = {}) {
-      const opts = Object.assign({}, sessionOptions);
+    if(timeOutMilliseconds < 0){
+      throw new Error('timeOut must be non-negative');
+    }else if(timeOutMilliseconds === 0 ){
```
Comment (steve-si) [inline Amp.js]: Not needed

File: Amp.js [hunk 4]
```diff
@@ -1,69 +1,68 @@
...
+      this.timeOut = timeOutMilliseconds;
+    }
+    this.timeOut = timeOutMilliseconds;
```
Comment (steve-si) [inline Amp.js]: `this.timeout = timeOutMilliseconds || 18000;`

File: Amp.js [hunk 5]
```diff
@@ -1,69 +1,68 @@
...
-      // resume
-      if (typeof sessionOptions === 'string') {
-        return _this.Session.deserialize(sessionOptions);
-      }
+    if(sessionLifeTimeSeconds < 0){
+      throw new Error('sessionLifeTime must be non-negative');
+    }else if (sessionLifeTimeSeconds === 0){
+      this.sessionLifeTime = 1800000; // 30 MINUTES
+    }
+    this.sessionLifeTime = 1000 * sessionLifeTimeSeconds;
```
Comment (steve-si) [inline Amp.js]: `this.sessionLifeTime = 1000 * sessionLifetimeSeconds || 30 * 60 * 1000;`

File: Amp.js [hunk 6]
```diff
@@ -1,69 +1,68 @@
...
+    }
+    this.timeOut = timeOutMilliseconds;
 
-      // resume
-      if (typeof sessionOptions === 'string') {
-        return _this.Session.deserialize(sessionOptions);
-      }
+    if(sessionLifeTimeSeconds < 0){
+      throw new Error('sessionLifeTime must be non-negative');
+    }else if (sessionLifeTimeSeconds === 0){
```
Comment (steve-si) [inline Amp.js]: Else block not needed.  See below.

**Review by steve-si — APPROVED** [review]

**Pull request merged**